### PR TITLE
[spec] Update algorithm

### LINKED ISSUE #289: algorithm.rst seems to need an update
the pseudo code there is still using old opcodes.

### PR DESCRIPTION
**Pull request by rossberg**
Fix #289.

### CODE REVIEW

**Comment by titzer** [comment]
lgtm

File: document/core/appendix/algorithm.rst [hunk 1]
```diff
@@ -212,22 +212,28 @@ Other instructions are checked in a similar manner.
          error_if(frame.opcode =/= if)
          push_ctrl(else, frame.start_types, frame.end_types)
 
-       case (try t1*->t2*)
+       case (try_table t1*->t2* handler*)
          pop_vals([t1*])
+         let rest = pop_vals()
```
Comment (aheejin) [inline document/core/appendix/algorithm.rst]: If `pop_vals` is called without a `types` parameter, does this mean it pops everything in there? (Is it documented somewhere?)

File: document/core/appendix/algorithm.rst [hunk 2]
```diff
@@ -212,22 +212,28 @@ Other instructions are checked in a similar manner.
...
+               push_val(Exnref)
+             case (catch_all)
+               skip
+             case (catch_all_ref)
+               push_val(Exnref)
+           error_if(ctrls.size() < handler.label)
+           pop_vals(label_types(ctrls[handler.label]))
+           error_if(not vals.is_empty())
+         push_vals(rest)
          push_ctrl(try, [t1*], [t2*])
```
Comment (aheejin) [inline document/core/appendix/algorithm.rst]:
```suggestion
         push_ctrl(try_table, [t1*], [t2*])
```

File: document/core/appendix/algorithm.rst [hunk 3]
```diff
@@ -25,7 +25,7 @@ Types are representable as an enumeration.
 
 .. code-block:: pseudo
 
-   type val_type = I32 | I64 | F32 | F64 | V128 | Funcref | Externref
+   type val_type = I32 | I64 | F32 | F64 | V128 | Funcref | Exnref | Externref
```
Comment (yamt) [inline document/core/appendix/algorithm.rst:28]: how about `is_ref` below? it seems unused though.

Comment (rossberg) [reply document/core/appendix/algorithm.rst:28]: Indeed, case added.

File: document/core/appendix/algorithm.rst (hunk 2 above)
Comment (rossberg) [reply document/core/appendix/algorithm.rst]: Fixed.

File: document/core/appendix/algorithm.rst (hunk 1 above)
Comment (rossberg) [reply document/core/appendix/algorithm.rst]: Good point, that was missing. I managed to change the algorithm to not require this function and be simpler at the same time.

**Review by yamt — APPROVED** [review]
looks reasonable to me

**Review by aheejin — APPROVED** [review]
Thanks!

**Pull request merged**